deepola/wake/src/processor/message_processor.rs

```diff
@@ -30,10 +30,11 @@ impl<T: Send, R: MessageProcessor<T> + Send> StreamProcessor<T> for R {
                     output_stream.write(message);
                     break;
                 }
-                Payload::Some(data_block) => {
-                    if let Some(df_acc) = self.process_msg(data_block.data()) {
-                        let message = DataMessage::from(df_acc);
-                        output_stream.write(message);
+                Payload::Some(dblock) => {
```

Comment:
This is one place where we are missing metadata. When there is an output from `post_process_msg()`, in that case the metadata needs to be supplied (the Payload::EOF scenario above).
(It is used in `q22`)

Reply:
Hmmm, good re-catch! I tried to find the metadata in `Payload::EOF`, but gave up & forgot about it. How do you recommend constructing the metadata? Cache last metadata and overwrite cardinality to 1.0?

Reply:
The last metadata should have cardinality as 1.0. I think just caching last metadata and cloning that should be fine for now.

Reply:
Cached and overwrote to 1.0 on EOF